tsl/src/remote/connection.c

```diff
@@ -1623,7 +1689,7 @@ TSConnection *
 remote_connection_open_session(const char *node_name, List *connection_options, bool set_dist_id)
 {
 	char *err = NULL;
-	TSConnection *conn = remote_connection_open(node_name, connection_options, &err);
+	TSConnection *conn = remote_connection_open(node_name, connection_options, TS_NO_TIMEOUT, &err);
```

Comment:
Is it ever a good idea to wait w/o timeout?

As an aside, the `TS_NO_TIMEOUT` define is pretty confusing, it's `DT_NOBEGIN`, which is in turn `PG_INT64_MIN`, which happens to match the `(long) -1` used by `WaitEventSetWait` on 64-bit systems, but not on 32-bit ones I guess? Does this lead to any problems?

Reply:
I think TS_NO_TIMEOUT is for TimestampTz type, not for the timeout itself, which is calculated based on it and has long type

Reply:
@akuzm Yes, it makes sense to wait without timeouts. Queries typically time out based on `statement_timeout` or interrupt by the user (ctrl-c). Both of these generate a signal that will trigger the latch, which is part of the wait set.